content/en/developers/dogstatsd/datagram_shell.md

```diff
@@ -86,11 +86,51 @@ The value is a Unix timestamp (UTC) and must be prefixed by `T`, for example:
 
 - `page.views:15|c|#env:dev|T1656581400`: A COUNT indicating that 15 page views happened on the 30th of June, 2022 at 9:30am UTC
 
+### DogStatsD protocol v1.4
+
+Starting with the Agent `>=v7.51.0`, a new inode value is supported for the container ID field.
+The container ID field can now contain two values to enrich DogStatsD metrics with additional container tags:
+- The container ID if available.
+- The cgroup node inode if the container ID is unavailable.
+
+The container ID field is still prefixed by `c:`, with the value being either:
+
+- `c:ci-<CONTAINER_ID>`
+- `c:in-<CGROUP_INODE>`
+
+### DogStatsD protocol v1.5
+
+Starting with the Agent `>=v7.57.0`, a new External Data field is supported.
+The Datadog Agent uses the External Data value to enrich DogStatsD metrics with additional container tags when the container ID is unavailable.
+
+The container ID is prefixed by `e:`, for example:
+
+`<METRIC_NAME>:<VALUE>|<TYPE>|#<TAG_KEY_1>:<TAG_VALUE_1>,<TAG_2>|e:<EXTERNAL_DATA>`
+
+This data is supplied by the [Datadog Agent Admission Controller][106] and will contain:
+- If the container is an init container or not.
+- The container name.
+- The pod UID.
+
+### DogStatsD protocol v1.6
+
+Starting with the Agent `>=v7.63.0`, a new Cardinality field is supported.
+The Datadog Agent uses the cardinality value to enrich DogStatsD metrics with additional container tags corresponding to their cardinality.
+
+The cardinality field is prefixed by `card:`, for example:
+
+`<METRIC_NAME>:<VALUE>|<TYPE>|#<TAG_KEY_1>:<TAG_VALUE_1>,<TAG_2>|card:<CARDINALITY>`
+
```

Comment:
This needs to list the available values for the cardinality field.

Reply:
Agreed, here they are: https://github.com/DataDog/documentation/pull/27076/files#diff-95284c3c410e9788783a6fe2e596a0c8a530c820e2d58bd23aa3614d07d4bcb6R139-R143